### PR TITLE
Implement FlashDecoding++ async softmax for split-K SDPA

### PR DESCRIPTION
**Pull request by Gasoonjia**
Replace online softmax (per-tile max tracking + cross-split rescaling) with a unified maximum value (phi=5.0) approach from FlashDecoding++.

Key changes:
- Split kernel: subtract fixed phi instead of tracking running max m_i, eliminating alpha rescaling between tiles
- Reduce kernel: simple summation of partial outputs instead of max-aware weighted combination; removes M_partial buffer
- ~12.9% average kernel-level speedup (6.8%-20.1% range) by saving HBM bandwidth (no M_partial reads/writes) and reducing ALU ops

The unified phi works because exp(qk - phi) is numerically stable for typical attention score ranges, and the fixed constant allows all splits to compute independently without synchronization.

Also used KernelAgent(https://github.com/meta-pytorch/KernelAgent) to further optimized the kernel.

Config | absolute perf (gain)
-- | --
p=128 d=128 | 149.9 (+9.7)
p=128 d=512 | 153.7 (+10.3)
p=256 d=128 | 149.6 (+9.5)
p=256 d=512 | 153.5 (+11.6)
p=512 d=128 | 149.1 (+9.1)
p=512 d=512 | 152.9 (+10.1)
p=1024 d=128 | 148.6 (+8.9)
p=1024 d=512 | 153.4 (+10.2)
p=2048 d=128 | 148.3 (+9.6)
p=2048 d=512 | 153.1 (+10.2)
Average | 151.2 (+9.9)

cc @digantdesai @freddan80 @per @zingo @oscarandersson8218 @mansnils @Sebastian-Larsson @robell

### CODE REVIEW

**Review by digantdesai — APPROVED** [review]
TIL FlashDecoding++, thanks.

Couple of things, seems like this is still online softmax (i.e you need to do reduce later). And can you add performance numbers with a sweep in the PR summary, thanks.

Also ask claude to do a quality comparisons using various prompts just to spotcheck quality.

**Comment by digantdesai** [comment]
@claude review this PR.

File: backends/cuda/triton/kernels/sdpa.py [hunk 1]
```diff
@@ -1175,15 +1175,10 @@ def _sdpa_decode_splitk_kernel(
                 mask_block, qk, tl.full(qk.shape, -float("inf"), dtype=tl.float32)
             )
 
-        # Online softmax update
-        m_ij = tl.maximum(m_i, tl.max(qk, axis=1).to(tl.float32))
-        safe_diff = tl.where(
-            m_ij[:, None] > -float("inf"), qk - m_ij[:, None], -float("inf")
-        )
+        # FlashDecoding++ async softmax: subtract unified phi instead of local max
```
Comment (Gasoonjia) [inline backends/cuda/triton/kernels/sdpa.py:1178]: @digantdesai here we replace the online softmax with async softmax by using a unified phi.

**Comment by Gasoonjia** [comment]
Update top comment with detailed perf comparsion.
